pkg/server/http/svc_test.go

```diff
@@ -0,0 +1,1123 @@
+	assert.Equal(t, 0, actual.GIDNumber, "GIDNumber doesn't match for user %v", expected.Username)
+
+}
+
+func deleteAccount(t *testing.T, id string) (*empty.Empty, error) {
+	client := service.Client()
+	cl := accountsProto.NewAccountsService("com.owncloud.api.accounts", client)
+
+	req := &accountsProto.DeleteAccountRequest{Id: id}
+	res, err := cl.DeleteAccount(context.Background(), req)
+	return res, err
+}
+
+func buildRoleServiceMock() settings.RoleService {
+	return settings.MockRoleService{
+		AssignRoleToUserFunc: func(ctx context.Context, req *settings.AssignRoleToUserRequest, opts ...client.CallOption) (res *settings.AssignRoleToUserResponse, err error) {
+			mockedRoleAssignment[req.AccountUuid] = req.RoleId
+			return &settings.AssignRoleToUserResponse{
+				Assignment: &settings.UserRoleAssignment{
+					AccountUuid: req.AccountUuid,
+					RoleId:      req.RoleId,
+				},
+			}, nil
+		},
+	}
+}
+
+func init() {
+	service = grpc.NewService(
+		grpc.Namespace("com.owncloud.api"),
+		grpc.Name("accounts"),
+		grpc.Address("localhost:9180"),
+	)
+
+	c := &accountsCfg.Config{
+		Server: accountsCfg.Server{
+			AccountsDataPath: dataPath,
+		},
+		Log: accountsCfg.Log{
+			Level:  "info",
+			Pretty: true,
+			Color:  true,
+		},
+	}
+
+	var hdlr *accountsSvc.Service
+	var err error
+
+	if hdlr, err = accountsSvc.New(
+		accountsSvc.Logger(accountsCmd.NewLogger(c)),
+		accountsSvc.Config(c),
+		accountsSvc.RoleService(buildRoleServiceMock())); err != nil {
+		log.Fatalf("Could not create new service")
+	}
+
+	hdlr.Client = mockClient{}
+
+	err = accountsProto.RegisterAccountsServiceHandler(service.Server(), hdlr)
+	if err != nil {
+		log.Fatal("could not register the Accounts handler")
+	}
+	err = accountsProto.RegisterGroupsServiceHandler(service.Server(), hdlr)
+	if err != nil {
+		log.Fatal("could not register the Groups handler")
+	}
+
+	err = service.Server().Start()
+	if err != nil {
+		log.Fatalf("could not start server: %v", err)
+	}
+}
+
+func cleanUp(t *testing.T) {
+	datastore := filepath.Join(dataPath, "accounts")
+
+	files, err := ioutil.ReadDir(datastore)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, f := range files {
+		found := false
+		for _, defUser := range DefaultUsers {
+			if f.Name() == defUser {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			deleteAccount(t, f.Name())
+		}
+	}
+}
+
+func sendRequest(method, endpoint, body, auth string) (*httptest.ResponseRecorder, error) {
+	var reader = strings.NewReader(body)
+	req, err := http.NewRequest(method, endpoint, reader)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	if auth != "" {
+		req.Header.Set("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(auth)))
+	}
+
+	rr := httptest.NewRecorder()
+
+	service := getService()
+	service.ServeHTTP(rr, req)
+
+	return rr, nil
+}
+
+func getService() svc.Service {
```

Comment:
unused method?

Reply:
Used in https://github.com/owncloud/ocis-ocs/pull/41/files#diff-872a4c7ddc55cabf71aac58455930020R303